### PR TITLE
[ENG-6310] Merge `develop` into GV feature branch and resolve conflicts

### PR DESCRIPTION
**Pull request by cslzchen**
## Ticket

https://openscience.atlassian.net/browse/ENG-6310

## Notes

Here is a record for what I did since the the conflict resolution got lost in the big merge commit. In case anything went wrong, we could look back here.

### The conflict file

* `addons/base/views.py`

```
➜  osf git:(feature/gravy_valet_integration) git merge upstream/develop
Auto-merging addons/base/views.py
CONFLICT (content): Merge conflict in addons/base/views.py
Auto-merging api/institutions/authentication.py
Auto-merging api/nodes/serializers.py
Auto-merging api/nodes/views.py
Auto-merging osf/models/mixins.py
Auto-merging osf/models/node.py
Auto-merging osf/models/preprint.py
Auto-merging osf/models/user.py
Auto-merging website/settings/defaults.py
Automatic merge failed; fix conflicts and then commit the result.
```

### The conflict

```
<<<<<<< HEAD
=======
def authenticate_via_oauth_bearer_token(resource, action):
    authorization = request.headers.get('Authorization')
    client = cas.get_client()

    try:
        access_token = cas.parse_auth_header(authorization)
        cas_resp = client.profile(access_token)
    except cas.CasError as err:
        sentry.log_exception(err)
        return json_renderer(err)  # Assuming json_renderer wraps the error in a Response

    permission = get_permission_for_action(action)
    if permission == permissions.READ:
        if resource.can_view_files():
            return OSFUser.load(cas_resp.user)
        required_scope = resource.file_read_scope
    else:
        required_scope = resource.file_write_scope

    token = cas_resp.attributes.get('accessTokenScope')
    if not token or not cas_resp.authenticated:
        raise HTTPError(http_status.HTTP_403_FORBIDDEN)

    normalize_scopes = oauth_scopes.normalize_scopes(cas_resp.attributes['accessTokenScope'])
    if required_scope not in normalize_scopes:
        raise HTTPError(http_status.HTTP_403_FORBIDDEN)

    return OSFUser.load(cas_resp.user)


def decrypt_and_decode_jwt_payload():
    try:
        payload_encrypted = request.args.get('payload', '').encode('utf-8')
        payload_decrypted = jwe.decrypt(payload_encrypted, WATERBUTLER_JWE_KEY)
        return jwt.decode(
            payload_decrypted,
            settings.WATERBUTLER_JWT_SECRET,
            options={'require': ['exp']},
            algorithms=[settings.WATERBUTLER_JWT_ALGORITHM],
        )['data']
    except (jwt.InvalidTokenError, KeyError) as err:
        sentry.log_message(str(err))
        raise HTTPError(http_status.HTTP_403_FORBIDDEN)


def get_authenticated_resource(resource_id):
    resource = AbstractNode.load(resource_id) or Preprint.load(resource_id)
    if not resource:
        raise HTTPError(http_status.HTTP_404_NOT_FOUND, message='Resource not found.')

    if resource.deleted:
        raise HTTPError(http_status.HTTP_410_GONE, message='Resource has been deleted.')

    if getattr(resource, 'is_retracted', False):
        raise HTTPError(http_status.HTTP_410_GONE, message='Resource has been retracted.')

    return resource


def _get_osfstorage_file_version(file_node: OsfStorageFileNode, version_string: str = None) -> FileVersion:
    if not (file_node and file_node.is_file):
        return None

    try:
        return FileVersion.objects.select_related('region').get(
            basefilenode=file_node,
            identifier=version_string or str(file_node.versions.count())
        )
    except FileVersion.DoesNotExist:
        raise HTTPError(http_status.HTTP_400_BAD_REQUEST, 'Requested File Version unavailable')


def _get_osfstorage_file_node(file_path: str) -> OsfStorageFileNode:
    if not file_path:
        return None

    file_id = file_path.strip('/')
    return OsfStorageFileNode.load(file_id)


def authenticate_user_if_needed(auth, waterbutler_data, resource):
    if auth.user:
        return  # User is already authenticated

    if 'cookie' in waterbutler_data:
        auth.user = OSFUser.from_cookie(waterbutler_data.get('cookie'))
        if not auth.user:
            raise HTTPError(http_status.HTTP_401_UNAUTHORIZED, 'Invalid user cookie.')

    authorization = request.headers.get('Authorization')
    if authorization and authorization.startswith('Bearer '):
        auth.user = authenticate_via_oauth_bearer_token(resource, waterbutler_data['action'])

>>>>>>> upstream/develop
```

### The Problem

* After some log digging, I found that the issue was: when an earlier version of GV was merged into develop, it created a few unexpected bugs; the bug was patched in `develop` while that piece of code was also refactored in GV.

### Solution

* Thus, simply removing this huge chunk of code should be good. All these methods have been deprecated or moved/renamed, which can be safely removed except for `get_authenticated_resource`.
  * `authenticate_via_oauth_bearer_token`:
    * Not found in `gv` and no usage found in `develop`, please double check
  * `decrypt_and_decode_jwt_payload`:
    * Refactored to [_decrypt_and_decode_jwt_payload](https://github.com/CenterForOpenScience/osf.io/blob/8484057ba5106593cbac7b1e94b4c9b82dece89d/addons/base/views.py#L257), looks the same
  * `get_authenticated_resource`:
    * Refactored to [_get_authenticated_resource](https://github.com/CenterForOpenScience/osf.io/blob/8484057ba5106593cbac7b1e94b4c9b82dece89d/addons/base/views.py#L272C5-L272C32). However, this one has a newer [patch](https://github.com/CenterForOpenScience/osf.io/blob/decc312a2be6111f1faccfd708d137a5d69343a7/addons/base/views.py#L320-L321) in `develop` that was not in `gv`, which was lost during merge conflicts resolving. I have patched it again in this PR.
  * `_get_osfstorage_file_version` and `_get_osfstorage_file_node`:
    * Refactored to [_get_osfstorage_file_version_and_node](https://github.com/CenterForOpenScience/osf.io/blob/8484057ba5106593cbac7b1e94b4c9b82dece89d/addons/base/views.py#L381C5-L381C42). This looks OK but may need another look.
  * `authenticate_user_if_needed`:
    * Not found in `gv` and no usage found in `develop`, please double check

### CODE REVIEW

**Review by brianjgeiger — APPROVED** [review]
Seems legit to me. I'll let @adlius do final review before merging it into the gvi branch.

**Review by adlius — APPROVED** [review]

**Pull request merged**
